### DISCUSSION TITLE
Listening for HTTP requests and displaying a count pattern

### DISCUSSION THREAD

**Discussion by raed667** [wxt-dev/wxt#466] [Q&A]
I'm building an extension that listen for HTTP requests in the background/worker and then displays the number of requests in the badge.

Sounds easy enough, but we have to keep a counter per tab. Also if the website in the tab changes, the counter needs to be reset to zero.

```ts
// background 
chrome.webRequest.onBeforeRequest.addListener((details) => {
  	// logic goes here
  },
  { urls: ['https://*.example.com/*'] },
  ['requestBody', 'extraHeaders']
);
```

I store the count in a global variable and then use `setBadgeText`, but i have to detect `tabs.onUpdated` and figure out if a URL change is real or an SPA etc.. and every day i discover an edge case. 

I feel like this should be something commun, am I missing something obvious ?

**Comment by aklinker1** [comment]
Not that I know of, sorry lol. Extension development can be tedious like this...

**Comment by raed667** [comment]
@aklinker1 ok sooo it turns out i was already handling every use-case except one. The worker shutting down. 

When the worker becomes inactive, the variable i was storing my count in is disposed of.

**Reply by aklinker1** [reply]
Yeah, you'll have to persist the count to storage, and probably use a mutex to make sure you don't mis-count.

**Reply by raed667** [reply]
or spam the `chrome.runtime.getPlatformInfo` function in an interval and prevent the worker from sleeping .. 🤷

**Reply by aklinker1** [reply]
Yup lol
